### PR TITLE
update installation and help docs

### LINKED ISSUE #123: Add some help text about determining your awk version
Getting the version number is different between different awk implementations.

Even though the track states that GNU awk is the version to be used and the
tests explicitly use `gawk`, people will ask.

I don't know where this should be documented.

* GNU awk
    ```
    $ awk --version
    GNU Awk 5.1.0, API: 3.0 (GNU MPFR 4.1.0, GNU MP 6.2.1)
    ...
    ```

* mawk
    ```
    $ sudo update-alternatives --config awk # => ... set mawk
    $ awk --version
    awk: not an option: --version
    $ awk -W version
    mawk 1.3.4 20200120
    ...
    ```

* MacOS default one-true-awk, aka "new awk"
    ```
    $ ( PATH=$(getconf PATH); awk --version )
    awk version 20200816
    ```

* busybox awk
    ```
    $ docker run -it alpine
    / # awk --version
    awk: unrecognized option: version
    BusyBox v1.35.0 (2022-05-09 17:27:12 UTC) multi-call binary.
    ...
    ```

### PR DESCRIPTION
**Pull request by glennj**
closes #123

### CODE REVIEW

File: docs/INSTALLATION.md [hunk 1]
```diff
@@ -1,17 +1,51 @@
 # Installation
 
+# What awk do I have?
+
+The Exercism AWK track uses `gawk`, GNU's AWK.
+You may already have it, let's see: execute this command
```
Comment (IsaacG) [inline docs/INSTALLATION.md:6]:
```suggestion
You may already have it; to check, execute this command
```

File: docs/INSTALLATION.md [hunk 2]
```diff
@@ -1,17 +1,51 @@
...
+```sh
+awk --version
+```
+
+If that returns:
+
+* something like `GNU Awk 5.1.0 ...`
+    - You've already got GNU awk, you can carry on with the exercises.
```
Comment (IsaacG) [inline docs/INSTALLATION.md:14]:
```suggestion
    - You already have GNU awk and can carry on with the exercises.
```

File: docs/INSTALLATION.md [hunk 3]
```diff
@@ -1,17 +1,51 @@
...
+
+* something like `awk version 20200816`
+    - This is is the "one true awk" implementation.
```
Comment (IsaacG) [inline docs/INSTALLATION.md:17]: Is this BSD awk?

File: docs/INSTALLATION.md [hunk 4]
```diff
@@ -1,17 +1,51 @@
...
+    - This is what ships with MacOS.
+    - Read below for installation instructions.
+
+* the error `awk: not an option: --version`
+    - You have `mawk` probably.
```
Comment (IsaacG) [inline docs/INSTALLATION.md:22]:
```suggestion
    - You probably have `mawk`.
```

File: docs/INSTALLATION.md [hunk 5]
```diff
@@ -1,17 +1,51 @@
...
+yum install gawk
+apk add gawk
 # and so on
 ```
+You may need to use `sudo` to execute the appropriate command.
 
 ## MacOS
 
-[Homebrew][brew] is your friend:
+[Homebrew][brew] is your friend: follow the link to install it and then
```
Comment (IsaacG) [inline docs/INSTALLATION.md:48]:
```suggestion
[Homebrew][brew] is your friend. Follow the link to install it and then run:
```

**Review by IsaacG — APPROVED** [review]

File: docs/INSTALLATION.md (hunk 3 above)
Comment (glennj) [reply docs/INSTALLATION.md:17]: I think so. I don't have one to test on except my Mac.

Comment (IsaacG) [reply docs/INSTALLATION.md:17]: I don't have Mac either, but I'm pretty sure it's BSD.

Comment (IsaacG) [reply docs/INSTALLATION.md:17]:
```suggestion
    - This is BSD awk.
```

**Review by IsaacG — APPROVED** [review]

**Pull request merged**